TEHIK_Open_Data_Loading_Scripts/main.py

```diff
@@ -14,64 +14,19 @@
 
 MANUAL_DATA = {
     "updatedOn": today[0],
-    "deceasedNumber": 125,
     "datesEnd": yesterday,
     "dates1Start": "2020-03-15",
     "dates2Start": "2020-02-26",
-    "intensive": [
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 4, 6, 7, 7,
-        7, 7, 10, 10, 13, 13, 15, 16, 16, 20, 17, 14, 12, 11,
-        9, 9, 11, 11, 9, 11, 10, 10, 11, 11, 10, 9, 8, 7,
-        7, 6, 6, 6, 7, 8, 10, 9, 7, 7, 7, 6, 6, 4,
-        4, 5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 4, 3, 2,
-        2, 2, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
-        1, 1, 1, 1, 1, 1, 2, 3, 3, 2, 3, 3, 3, 2,
-        2, 1, 0, 0, 0, 1, 1, 2, 2, 2, 3, 3, 4, 3,
-        2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1,
-        1, 1, 3, 3, 4, 3, 1, 1, 3, 2, 2, 2, 2, 2,
-        4, 4, 4, 4, 4, 5, 6, 6, 5, 5, 5, 7, 6, 6,
-        7, 10, 8, 7, 8, 8, 7, 7, 6, 6, 7, 11, 12, 11,
-        12, 11
-    ],
-    "deceased": [
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
-        1, 1, 1, 3, 3, 4, 5, 11, 12, 13, 15, 19, 21, 24,
-        24, 24, 24, 25, 27, 31, 35, 36, 38, 38, 40, 40, 43, 44,
-        45, 46, 47, 50, 52, 52, 52, 54, 54, 55, 57, 57, 57, 57,
-        59, 59, 60, 60, 61, 61, 61, 62, 63, 63, 63, 64, 64, 64,
-        64, 64, 64, 64, 65, 65, 66, 66, 67, 67, 68, 68, 68, 69,
-        69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69,
-        69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69,
-        69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69,
-        69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69,
-        69, 69, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63,
-        63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 64, 64, 64,
-        64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-        64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
-        64, 64, 64, 64, 64, 64, 64, 65, 66, 67, 67, 67, 67, 67,
-        67, 68, 68, 68, 68, 68, 68, 68, 68, 68, 68, 68, 71, 71,
-        71, 73, 73, 73, 73, 73, 73, 73, 73, 73, 73, 73, 73, 73,
-        73, 73, 73, 75, 76, 76, 76, 76, 80, 80, 81, 81, 84, 85,
-        86, 87, 87, 88, 92, 94, 97, 99, 104, 109, 112, 118, 121, 122,
-        123, 125
-    ]
-}
+    }
 
 
 ######## CONFIGURE IO LOCATIONS ########
 
 API_ENDPOINT = "https://opendata.digilugu.ee/opendata_covid19_test_results.json"
 MUNICIPALITIES_ENDPOINT = "https://opendata.digilugu.ee/opendata_covid19_test_location.json"
 HOSPITAL_ENDPOINT = "https://opendata.digilugu.ee/opendata_covid19_hospitalization_timeline.json"
+MANUAL_DATA_FILE_LOCATION = "manualData.json"
```

Comment:
Camel case isn't recommended in Python, so I would call the manual data file: "manual_data.json", which is then also consistent with the other JSON endpoints.

Reply:
My proposal would be that we schedule the cron job to check for updates every 15 minutes after 10.30am. The first thing we should do is to check whether the data has already been updated for the current day, and if so exit immediately, otherwise continue with the process to download any data from TEHIK and process it. After downloading the data from TEHIK, we need to add a check that the data is at least one day after the most recent update we have processed, and if not exit.

In addition to these checks, the simplest way to implement error handling in relation to the update process would be to put the entire process from line 44 onwards of main.py in a "try" block. In the "except" block, we should log the error, ideally in a way can be shared with Slack as you suggested. If we import the "traceback" module, then we can get the details of the exception which caused the error with:

```
exception_details = traceback.format_exc()
```

Instead of scheduling a separate cron job to scrape the deaths, I would call the function to scrape the deaths from the main update function.

Overall, this would mean the data file will only be updated if the entire process goes without errors, and it ensures that the data will be updated relatively soon after it is available without making too many unnecessary calls to the data at TEHIK.

Before sending this code live, we will need to incorporate any recent manual data updates.